socorro/external/crashstorage_base.py

```diff
@@ -373,55 +365,55 @@ def ack_crash(self, crash_id):
 
 
 class PolyStorageError(Exception, collections.MutableSequence):
-    """an exception container holding a sequence of exceptions with tracebacks.
+    """Exception container holding a sequence of exceptions with tracebacks
+
+    :arg message: an optional over all error message
 
-    parameters:
-        message - an optional over all error message
     """
     def __init__(self, *args):
         super(PolyStorageError, self).__init__(*args)
         self.exceptions = []  # the collection
 
     def gather_current_exception(self):
-        """append the currently active exception to the collection"""
+        """Append the currently active exception to the collection"""
         self.exceptions.append(sys.exc_info())
 
     def has_exceptions(self):
-        """the boolean opposite of is_empty"""""
+        """Boolean opposite of is_empty"""""
         return bool(self.exceptions)
 
     def __len__(self):
-        """how many exceptions are stored?
-        this method is required by the MutableSequence abstract base class"""
+        """Length of exception collection"""
         return len(self.exceptions)
 
     def __iter__(self):
-        """start an iterator over the squence.
-        this method is required by the MutableSequence abstract base class"""
+        """Start an iterator over the sequence"""
         return iter(self.exceptions)
 
     def __contains__(self, value):
-        """search the sequence for a value and return true if it is present
-        this method is required by the MutableSequence abstract base class"""
-
+        """Search the sequence for a value and return true if it is present"""
         return self.exceptions.__contains__(value)
 
     def __getitem__(self, index):
-        """fetch a specific exception
-        this method is required by the MutableSequence abstract base class"""
+        """Get a specific exception by index"""
         return self.exceptions.__getitem__(index)
 
     def __setitem__(self, index, value):
-        """change the value for an index in the sequence
-        this method is required by the MutableSequence abstract base class"""
+        """Set the value for an index in the sequence"""
         self.exceptions.__setitem__(index, value)
 
     def __str__(self):
         output = []
         if self.args:
             output.append(self.args[0])
         for e in self.exceptions:
-            output.append(repr(e[1]).encode('ascii', 'ignore'))
+            output.append(repr(e[1]))
+        if six.PY2:
+            # If this is Python2, encode everything into strings before joining
+            output = [
+                part.encode('ascii', 'ignore')
+                for part in output
+            ]
```

Comment:
In Python 2, some of these are unicode. So we convert everything to ascii-encoded strings before we can join via a string.

In Python 3, there ain't no problems here.